cv_bridge/python/cv_bridge/core.py

```diff
@@ -90,6 +95,46 @@ def encoding_to_cvtype2(self, encoding):
     def encoding_to_dtype_with_channels(self, encoding):
         return self.cvtype2_to_dtype_with_channels(self.encoding_to_cvtype2(encoding))
 
+    def compressed_imgmsg_to_cv2(self, cmprs_img_msg, desired_encoding = "passthrough"):
+        """
+        Convert a sensor_msgs::CompressedImage message to an OpenCV :cpp:type:`cv::Mat`.
+
+        :param img_msg:   A :cpp:type:`sensor_msgs::Image` message
+        :param desired_encoding:  The encoding of the image data, one of the following strings:
+
+           * ``"passthrough"``
+           * one of the standard strings in sensor_msgs/image_encodings.h
+
+        :rtype: :cpp:type:`cv::Mat`
+        :raises CvBridgeError: when conversion is not possible.
+
+        If desired_encoding is ``"passthrough"``, then the returned image has the same format as img_msg.
+        Otherwise desired_encoding must be one of the standard image encodings
+
+        This function returns an OpenCV :cpp:type:`cv::Mat` message on success, or raises :exc:`cv_bridge.CvBridgeError` on failure.
+
+        If the image only has one channel, the shape has size 2 (width and height)
+        """
+        import cv2
+        import numpy as np
+
+        str_msg = cmprs_img_msg.data
+        buf = np.ndarray(shape=(1, len(str_msg)),
+                          dtype=np.uint8, buffer=cmprs_img_msg.data)
+        im = cv2.imdecode(buf, cv2.IMREAD_ANYCOLOR)
+
+        if desired_encoding == "passthrough":
+            return im
+
+        from cv_bridge.boost.cv_bridge_boost import cvtColor2
+
+        try:
+            res = cvtColor2(im, "bgr8", desired_encoding)
```

Comment:
Actually, we do not know the original encoding of the image. Let's say it is a JPG with 3 channels. Maybe it is rgb8 or bgr8, or just 8UC3. I'd say you should just remove the desired_encoding and assume it is always passthrough. People can then have their logic outside of the function to convert or not.